### DISCUSSION TITLE
[Help] Canvas VS SVG (not sure if it's a bug)

### DISCUSSION THREAD

**Discussion by faizulhaque** [fabricjs/fabric.js#9177] [Q&A] [answered]
I am trying to add image inside Polygon shape, and it works great on Canvas, however, when I can generate the SVG, it's not respecting the design

Reference:
https://jsfiddle.net/faizulhaque124/jd5xtsqy/1/

P.S: I tried the same logic using latest fabricjs, but seems like the clipTo function is deprecated.

**Comment by ShaMan123** [comment, marked as answer]
I didn't look closely into your repro however it seems you have a custom rendering function but you didn't override the svg methods accordingly (`_toSVG` or `toSVG`)
If you do not need to run the svg output in the browser I consider using fabric in node because then you can utilize the built in svg output in node-canvas
